api/search.go

```diff
@@ -625,8 +609,8 @@ func AddNlpToSearch(ctx context.Context, queryBuilder QueryBuilder, params url.V
 
 	// If berlin exists, add the subdivisions to NLP criteria.
 	// They'll be used later in the query to ElasticSearch
-	if berlin != nil && len(berlin.Matches) > 0 && len(berlin.Matches[0].Subdivision) == 2 {
-		nlpCriteria = queryBuilder.AddNlpSubdivisionSearch(nlpCriteria, berlin.Matches[0].Subdivision[1])
+	if len(berlin.Matches) > 0 && len(berlin.Matches[0].Loc.Subdivision) == 2 {
```

Comment:
You've removed a layer of protection here that I put in to stop this line falling over.

Reply:
Because of lines 561-573 I made sure that berlin is never NIL as it's query field is required in the category call.